components/plurals/src/rules/runtime/resolver.rs

```diff
@@ -0,0 +1,61 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::operands::PluralOperands;
+use crate::rules::runtime::ast;
+
+pub fn test_rule(rule: &ast::Rule, operands: &PluralOperands) -> bool {
```

Comment:
Suggestion: I suggested this algorithm for resolving the and/or rules, but it is non-obvious.  I think it would be good to add some prose explaining it.